### ISSUE TITLE
Java build stuck with blazer persistence framework

### ISSUE THREAD

**Issue by bbobbettoS** [Blazebit/blaze-persistence#1551]
When I try to build an existent maven java project where I have added the blaze framework in it, the build process stuck. 

My Steps to reproduce
In a fresh 2022.09 eclipse installation, 
- download the entityView maven archetype
'mvn archetype:generate "-DarchetypeGroupId=com.blazebit" "-DarchetypeArtifactId=blaze-persistence-archetype-entity-view-sample" "-DarchetypeVersion=1.6.7"'
- Imports as "Existing Maven Project"
- Disable "build automatically"
- Add jdk1.8 to "installed JREs" and sets as default
- Uncheck "Use compliance from execution..." and "Use default compliance settings" from "JDK Compliance"
- "Apply and Close" and rebuild

This steps are to reproduce the problem, in my case I am using jdk11 and I not have unchecked "Use compliance from execution..." and "Use default compliance settings", but after I added blaze libs and created a few Blaze interfaces, my project doesn't build anymore.
When I switch to the branch pre-blaze I can build without any problem

## What is the current bug behavior?
On the bottom Rebuilt is stuck at X% and I can only kill eclipse because all the other operation, closing included, will wait the rebuild process to finish.
On restart "Use compliance from execution..." and "Use default compliance settings" are checked, but every time I try to build the project, the build process stuck on 5%

## Priority

- [ ] Urgent
- [x] High
- [ ] Medium
- [ ] Low

## Severity

- [x] Blocker
- [ ] Major
- [ ] Normal
- [ ] Low

![eclipse](https://user-images.githubusercontent.com/114412356/192487603-77245db8-0818-428b-b4a7-ec049932d42a.gif)

**Comment by bbobbettoS** [comment]
After jstack the Building worker is stuck
'''
"Worker-13: Building" #66 prio=5 os_prio=0 cpu=304,21ms elapsed=256,54s tid=0x00007ff3d22a42a0 nid=0x10cc waiting on condition  [0x00007ff338bfd000]
   java.lang.Thread.State: WAITING (parking)
	at jdk.internal.misc.Unsafe.park(java.base@17.0.4.1/Native Method)
	- parking to wait for  <0x00000000a4a93dd0> (a java.util.concurrent.FutureTask)
	at java.util.concurrent.locks.LockSupport.park(java.base@17.0.4.1/LockSupport.java:211)
	at java.util.concurrent.FutureTask.awaitDone(java.base@17.0.4.1/FutureTask.java:447)
	at java.util.concurrent.FutureTask.get(java.base@17.0.4.1/FutureTask.java:190)
	at com.blazebit.persistence.view.processor.EntityViewAnnotationProcessor.await(EntityViewAnnotationProcessor.java:145)
	at com.blazebit.persistence.view.processor.EntityViewAnnotationProcessor.createMetaModelClasses(EntityViewAnnotationProcessor.java:234)
	at com.blazebit.persistence.view.processor.EntityViewAnnotationProcessor.execute(EntityViewAnnotationProcessor.java:125)
	at com.blazebit.persistence.view.processor.EntityViewAnnotationProcessor.process(EntityViewAnnotationProcessor.java:99)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:142)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:124)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:172)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:138)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:953)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:450)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:426)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:379)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:214)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:311)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:79)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:273)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:211)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:1024)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:254)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:311)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:400)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:403)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:514)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:462)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:544)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:161)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:255)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

   Locked ownable synchronizers:
	- None
'''

**Comment by beikov** [comment]
Thanks a lot. Can you share the stack traces of the other threads as well?

**Comment by bbobbettoS** [comment]
Yes, sorry
[eclipse_stack.txt](https://github.com/Blazebit/blaze-persistence/files/9654072/eclipse_stack.txt)
If can be useful this is the stack after 1 hour of build stuck at 64%
[eclipse_stack2.txt](https://github.com/Blazebit/blaze-persistence/files/9654083/eclipse_stack2.txt)

**Comment by beikov** [comment]
As a workaround, one thing you can do is to disable parallelization in the annotation processor, by passing the annotation processor argument `threads=1`. I think this [post](https://stackoverflow.com/questions/43404891/how-to-configure-java-annotation-processors-in-eclipse) should be able to help you understand how to configure this.

**Comment by beikov** [comment]
As far as I can tell from the thread stacks, it seems like this could be a bug in Eclipse. An annotation processor thread tries to close a stream which then waits in eclipse specific code for a lock, which it never gets. I can only assume that this has to do with parallelization i.e. if the main annotation processor thread would try to acquire the lock, then it would get that.

I have played with the thought of rewriting the annotation processor architecture anyway since the parallelization runs into certain issues on newer JDKs. Let me know if the `threads=1` workaround fixes your issue. For further performance improvements of the annotation processor, you can follow https://github.com/Blazebit/blaze-persistence/issues/1547. Since you are just starting to use Blaze-Persistence, the parallelization will probably not make a big difference anyway. The big improvements start when compiling a workspace with 500+ entity views.

**Comment by bbobbettoS** [comment]
I did
![proc_ann](https://user-images.githubusercontent.com/114412356/192493659-4b71cc9e-c873-48b9-9832-bf37a69966d0.gif)
Rebuilding is stuck to 14%
[eclipse_7224.txt](https://github.com/Blazebit/blaze-persistence/files/9654245/eclipse_7224.txt)
I don't known if it is related, but even vscode doesn't build anymore, I think that it uses the same/similar jdt plugin

**Comment by beikov** [comment]
If VSCode also uses the Eclipse JDT compiler, then it will probably run into the same issues. The latest thread stack dump still shows that multi-threading is used, so I don't know if Eclipse properly passes the options to the annotation processor, but I think you might still run into the problem even after doing that, since a single thread executor will still run in a separate thread. I'll make sure that the `threads=1` option will run in the same thread.

**Comment by beikov** [comment]
Can you try building [this PR](https://github.com/Blazebit/blaze-persistence/pull/1552) and see if that fixes your issues? Given that you can get Eclipse to pass on the annotation processor configuration properly.

**Comment by bbobbettoS** [comment]
Thank you, the fix has resolved this issue, in yours "Entity view archetype" and "Spring-Boot archetype" test projects, but I've encountered another issue that stuck the build process while integrating blaze persistence in a spring boot project of mine.
This happens with both versions (1.6.7 and 1.6.8-SNAPSHOT)
This is the thread dump
[spring_boot.txt](https://github.com/Blazebit/blaze-persistence/files/9655881/spring_boot.txt)

**Comment by beikov** [comment]
Well, it blocks the same way as before, so maybe your annotation processor configuration of `threads=1` doesn't get applied correctly on that other project?

**Comment by beikov** [comment]
So this issue is fixed with the PR I created?

**Comment by bbobbettoS** [comment]
Yes, I have tested again, with this configuration
![immagine](https://user-images.githubusercontent.com/114412356/192582101-8eb5ae44-73aa-4812-9086-3109a629abbe.png)
and your PR my spring boot project compile and run
Thank you.

**Comment by beikov** [comment]
Could you please try to build the https://github.com/Blazebit/blaze-persistence/pull/1555 PR and let me know if it works for you without the `thread=1` workaround?

**Comment by bbobbettoS** [comment]
Hello, 
I confirm that both project (your maven test project "Entity view archetype" and mine) compile successfully with #1555 PR and without `thread=1` workaround.
Thank you.

**Comment by beikov** [comment]
Thanks so much. I hope I can release version 1.6.8 with this fix this week. I'll ping you when it's available on Maven Central
